src/main/java/com/dell/isg/smi/service/device/discovery/manager/DiscoveryManagerImpl.java

```diff
@@ -176,7 +172,7 @@
             if (EnumUtils.isValidEnum(DiscoveryDeviceGroupEnum.class, discoverGroupName) && !StringUtils.equals(discoverGroupName, DiscoveryDeviceTypeEnum.UNKNOWN.value())) {
                 DiscoveryDeviceGroupEnum discoverGroup = DiscoveryDeviceGroupEnum.valueOf(discoverGroupName);
                 if (identifyDeviceType((List<DiscoveredDeviceInfo>) CollectionUtils.select(discoverDeviceInfos, predicateReachableUndiscoveredDevice()), discoverGroup)) {
-                    for (String deviceName : discoveryDeviceConfigProvider.getAllDeviceNameByGroup(discoverGroup)) {
+                    for (String deviceName : discoveryDeviceConfigProvider.getAllDeviceTypeNameByGroup(discoverGroup)) {
```

Comment:
Break into it's own variable of device names. Also possible NPE - what does sonar say?